include/seqan3/argument_parser/detail/format_base.hpp

```diff
@@ -371,49 +352,59 @@ class format_help_base : public format_base
 
     /*!\brief Adds a print_section call to parser_set_up_calls.
-     * \param[in] title The title of the section of the help page.
+     * \copydetails seqan3::argument_parser::add_section
      */
-    void add_section(std::string const & title)
+    void add_section(std::string const & title, option_spec const spec)
     {
-        parser_set_up_calls.push_back([this, title] ()
+        if (!(spec & option_spec::HIDDEN) && (!(spec & option_spec::ADVANCED) || show_advanced_options))
         {
-            derived_t().print_section(title);
-        });
+            parser_set_up_calls.push_back([this, title] ()
+            {
+                derived_t().print_section(title);
+            });
+        }
     }
 
     /*!\brief Adds a print_subsection call to parser_set_up_calls.
-     * \param[in] title The title of the subsection of the help page.
+     * \copydetails seqan3::argument_parser::add_subsection
      */
-    void add_subsection(std::string const & title)
+    void add_subsection(std::string const & title, option_spec const spec)
     {
-        parser_set_up_calls.push_back([this, title] ()
+        if (!(spec & option_spec::HIDDEN) && (!(spec & option_spec::ADVANCED) || show_advanced_options))
         {
-            derived_t().print_subsection(title);
-        });
+            parser_set_up_calls.push_back([this, title] ()
+            {
+                derived_t().print_subsection(title);
+            });
+        }
     }
 
     /*!\brief Adds a print_line call to parser_set_up_calls.
-     * \param[in] text The line text to be printed to the help page.
-     * \param[in] line_is_paragraph True if you want a new line at the end.
+     * \copydetails seqan3::argument_parser::add_line
      */
-    void add_line(std::string const & text, bool line_is_paragraph)
+    void add_line(std::string const & text, bool line_is_paragraph, option_spec const spec)
     {
-        parser_set_up_calls.push_back([this, text, line_is_paragraph] ()
+        if (!(spec & option_spec::HIDDEN) && (!(spec & option_spec::ADVANCED) || show_advanced_options))
         {
-            derived_t().print_line(text, line_is_paragraph);
-        });
+            parser_set_up_calls.push_back([this, text, line_is_paragraph] ()
+            {
+                derived_t().print_line(text, line_is_paragraph);
+            });
+        }
     }
 
     /*!\brief Adds a seqan3::print_list_item call to parser_set_up_calls.
-     * \param[in] key The key of the key-value pair list item.
-     * \param[in] desc The key of the key-value pair list item.
+     * \copydetails seqan3::argument_parser::add_list_item
      */
-    void add_list_item(std::string const & key, std::string const & desc)
+    void add_list_item(std::string const & key, std::string const & desc, option_spec const spec)
     {
-        parser_set_up_calls.push_back([this, key, desc] ()
+        if (!(spec & option_spec::HIDDEN) && (!(spec & option_spec::ADVANCED) || show_advanced_options))
```

Comment:
This is kind of repeating itself. Thus, it would be good to put it into its own private function in order to avoid disambiguities in case the if statement needs to be updated or a field is added.

Reply:
A comment about what the expression means would be then helpful.

Reply:
Good point. I put this into a function and documented it properly.
Feel free to make a name suggestion for the function as I had troubles finding one..

Reply:
`[store|register]_help_page_element`?